src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -157,6 +157,8 @@ protected static Object invokeMethod(Method thisMethod, Object instance, Object[
         parameters.add(method);
       } else if (cls.equals(ITestContext.class)) {
         parameters.add(testContext);
+      } else if (cls.equals(Class.class)) {
+        parameters.add(com.getDeclaringClass());

```

Comment:
What's the use case for this? Data providers are typically non static so they already have access to the declaring class.